tests/parshaping-simple.sil

```diff
@@ -0,0 +1,89 @@
+  { width = SILE.measurement("158.09926166671pt"), indent=SILE.measurement("88.782369166646pt") },
+  { width = SILE.measurement("164.08698356407pt"), indent=SILE.measurement("85.788508217965pt") },
+  { width = SILE.measurement("167.09452358097pt"), indent=SILE.measurement("84.284738209513pt") },
+  { width = SILE.measurement("167.28269918420pt"), indent=SILE.measurement("84.190650407902pt") },
+  { width = SILE.measurement("164.66117660805pt"), indent=SILE.measurement("85.501411695975pt") },
+  { width = SILE.measurement("159.09112080046pt"), indent=SILE.measurement("88.286439599772pt") },
+  { width = SILE.measurement("150.24495450145pt"), indent=SILE.measurement("92.709522749275pt") },
+  { width = SILE.measurement("137.49177425846pt"), indent=SILE.measurement("99.086112870769pt") },
+  { width = SILE.measurement("119.58808312009pt"), indent=SILE.measurement("108.03795843996pt") },
+  { width = SILE.measurement("93.624309129279pt"), indent=SILE.measurement("121.01984543536pt") },
+  { width = SILE.measurement("47.922780548840pt"), indent=SILE.measurement("143.87060972558pt") },
+  { width = SILE.measurement("167.54343340503pt"), indent=SILE.measurement("84.060283297483pt") }
+}
+
+SILE.registerCommand("testparshaping", function (options, content)
+  local oldSetup = SILE.linebreak.setupLineLengths
+
+  -- Ship out previous content and start a new paragraph.
+  SILE.typesetter:leaveHmode()
+
+  SILE.settings.temporarily(function()
+    -- Ensure reasonably fixed baselineskip, and some more tolerance
+    -- as circles are hard.
+    SILE.settings.set("linebreak.tolerance", 2000)
+    SILE.settings.set("document.baselineskip", "1.2em")
+    SILE.settings.set("document.parindent", 0)
+
+    -- Override linebreak setup to implement our parshape
+    SILE.linebreak.setupLineLengths = function (self)    
+        self.parShape = function (self, line)
+            if circle[line] then
+                return {
+                    width = circle[line].width,
+                    left = circle[line].indent:tonumber(),
+                    right = circle[line].indent:tonumber()
+                }
+            end
+            -- Paragraphs that extend beyond the circle...
+            return {
+                width = self.hsize,
+                left = 0,
+                right = 0
+            }
+        end 
+        -- Assume we should never reach 500 lines. If it does, stop
+        -- the mess.
+        self.lastSpecialLine = 500
+        self.secondWidth = self.hsize
+        self.easy_line = self.lastSpecialLine -- implies looseness=0
+    end
+    SILE.process(content)
+    SILE.typesetter:leaveHmode();
+
+    -- Restore the original linebreaks.
+    SILE.linebreak.setupLineLengths = oldSetup
+    -- Also do not forget to deactivate the parShape
+    SILE.linebreak.parShape = nil
+  end)
+end, "Format paragraphs in a fixed centered circle")
+}
+
+Galileo, in 1638, wrote:
```

Comment:
@alerque also taught me not to put unnecessary paragraphs in tests. :-)

Reply:
Done.

Reply:
Yup, the less content the better. The more content you have the more change future versions of fonts or other unrelated changes will kick in. Testing the minimum possible surface area for a given feature or bug is the most maintainable way. Even now and with a bunch of custom tooling on my side (such as iterating through visual diffs of outputs) updating tests when things change is a significant pain point.